spec/controllers/followup_controller_spec.rb

```diff
@@ -0,0 +1,255 @@
+                         :incoming_message_id => message_id
+      expect(response).to render_template('user/wrong_user')
+    end
+
+    it "does not allow follow ups to external requests" do
+      session[:user_id] = FactoryGirl.create(:user)
+      external_request = FactoryGirl.create(:external_request)
+      get :new, :request_id => external_request.id
+      expect(response).to render_template('followup_bad')
+      expect(assigns[:reason]).to eq('external')
+    end
+
+    it "redirects to the signin page if not logged in" do
+      get :new, :request_id => request.id
+      expect(response).
+        to redirect_to(signin_url(:token => get_last_post_redirect.token))
+    end
+
+    it "calls the message a followup if there is an incoming message" do
+      expected_reason = "To send a follow up message to #{request.public_body.name}"
+      get :new, :request_id => request.id, :incoming_message_id => message_id
+       expect(get_last_post_redirect.reason_params[:web]).to eq(expected_reason)
+    end
+
+    it "calls the message a reply if there is no incoming message" do
+      expected_reason = "To reply to #{request.public_body.name}."
+      get :new, :request_id => request.id
+      expect(get_last_post_redirect.reason_params[:web]).to eq(expected_reason)
+    end
+
+    context "logged in as the request owner" do
+
+      before(:each) do
+        session[:user_id] = request_user.id
+      end
+
+      it "shows the followup form" do
+        get :new, :request_id => request.id, :incoming_message_id => message_id
+        expect(response).to render_template('new')
+      end
+
+      it "offers the opportunity to reply to the main address" do
+        get :new, :request_id => request.id, :incoming_message_id => message_id
+        expect(response.body).
+          to have_css("div#other_recipients ul li", :text => "the main FOI contact address for")
+      end
+
+      it "offers an opportunity to reply to another address" do
+        open_request = FactoryGirl.create(:info_request_with_incoming,
+                                          :user => request_user,
+                                          :allow_new_responses_from => "anybody")
+        receive_incoming_mail('incoming-request-plain.email',
+                              open_request.incoming_email, "Frob <frob@bonce.com>")
+        get :new, :request_id => open_request.id,
+                           :incoming_message_id => open_request.incoming_messages[0].id
+        expect(response.body).to have_css("div#other_recipients ul li", :text => "Frob")
+      end
+
+      context "the request is hidden" do
+
+        let(:hidden_request) do
+          FactoryGirl.create(:info_request_with_incoming, :user => request_user,
+                                                          :prominence => "hidden")
+        end
+
+        it "does not show the form, even to the request owner" do
+          get :new, :request_id => hidden_request.id
```

Comment:
I previously [commented](https://github.com/mysociety/alaveteli/commit/6a3c8b27694c8f165be68c88d422b6bcc36cf670#commitcomment-15479158) that the [original specs](https://github.com/mysociety/alaveteli/blob/b248af79eecc6a2ff52f603c2e7cd6961733fb23/spec/controllers/response_controller_spec.rb#L39-L60) were passing an `:incoming_message_id` parameter. Your [comment](https://github.com/mysociety/alaveteli/commit/6a3c8b27694c8f165be68c88d422b6bcc36cf670#commitcomment-15704128) says that it was reinstated, but doesn't appear to be here.